# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/gtest/TestAudioInputSource.cpp
# Commit: f81e1b1615b3
# Full Hash: f81e1b1615b3a9c5ef6c62a291f79029b3807dde
# Author: Karl Tomlinson <karlt+@karlt.net>
# Date: 2023-03-31 09:51:00
# Regressor Bug: 1801190
# File Overlap Count: 1
# Description:
#   Bug 1801190 do not reset mAudioStreamState to None on state changes from an old cubeb_stream r=pehrsons
#   
#   Such resets were preventing transitions to Running for new streams and allowing
#   MaybeStartAudioStream() to start another stream.
#   
# ==============================================================================

diff -r 9592d9865f5a -r f81e1b1615b3 dom/media/gtest/TestAudioInputSource.cpp
--- a/dom/media/gtest/TestAudioInputSource.cpp	Fri Mar 31 02:23:14 2023 +0300
+++ b/dom/media/gtest/TestAudioInputSource.cpp	Thu Mar 30 23:18:44 2023 +0000
@@ -54,7 +54,7 @@
   EXPECT_CALL(*listener,
               AudioStateCallback(
                   sourceId, AudioInputSource::EventListener::State::Stopped))
-      .Times(2);
+      .Times(4);
 
   RefPtr<AudioInputSource> ais = MakeRefPtr<AudioInputSource>(
       std::move(listener), sourceId, deviceId, channels, true, testPrincipal,
@@ -118,7 +118,8 @@
                   sourceId, AudioInputSource::EventListener::State::Started));
   EXPECT_CALL(*listener,
               AudioStateCallback(
-                  sourceId, AudioInputSource::EventListener::State::Stopped));
+                  sourceId, AudioInputSource::EventListener::State::Stopped))
+      .Times(2);
 
   RefPtr<AudioInputSource> ais = MakeRefPtr<AudioInputSource>(
       std::move(listener), sourceId, deviceId, channels, true, testPrincipal,
@@ -198,6 +199,9 @@
   EXPECT_CALL(*listener,
               AudioStateCallback(
                   sourceId, AudioInputSource::EventListener::State::Error));
+  EXPECT_CALL(*listener,
+              AudioStateCallback(
+                  sourceId, AudioInputSource::EventListener::State::Stopped));
 
   RefPtr<AudioInputSource> ais = MakeRefPtr<AudioInputSource>(
       std::move(listener), sourceId, deviceId, channels, true, testPrincipal,
@@ -245,7 +249,8 @@
                   sourceId, AudioInputSource::EventListener::State::Started));
   EXPECT_CALL(*listener,
               AudioStateCallback(
-                  sourceId, AudioInputSource::EventListener::State::Stopped));
+                  sourceId, AudioInputSource::EventListener::State::Stopped))
+      .Times(2);
 
   RefPtr<AudioInputSource> ais = MakeRefPtr<AudioInputSource>(
       std::move(listener), sourceId, deviceId, channels, true, testPrincipal,